src/statsig.erl

```diff
@@ -1,140 +1,48 @@
 -module(statsig).
 
--behaviour(gen_server).
+-behaviour(application).
 
 -export(
   [
-    init/1,
-    terminate/2,
-    handle_call/3,
-    handle_cast/2,
-    handle_info/2,
-    check_gate/3,
+    start_link/0,
+    start/2,
+    stop/1,
+    check_gate/2,
+    log_event/3,
     log_event/4,
-    log_event/5,
-    flush/1
+    flush/0
   ]
 ).
 
--import(utils, [get_timestamp/0]).
--import(network, [start/0, stop/0, request/4]).
--import(evaluator, [eval_gate/3]).
--import(logging, [get_event/4, get_exposure/4]).
+start_link() ->
```

Comment:
I think the `start_link` doesn't make sense here, since the module is implementing the `application` behaviour.

It errors out as follows:
```erlang
2> statsig:start_link().
{error,{undef,[{statsig,init,[[]],[]},
               {gen_server,init_it,2,[{file,"gen_server.erl"},{line,423}]},
               {gen_server,init_it,6,[{file,"gen_server.erl"},{line,390}]},
               {proc_lib,init_p_do_apply,3,
                         [{file,"proc_lib.erl"},{line,226}]}]}}
```